plugin/http/httptrace/httptrace.go

```diff
@@ -59,10 +63,75 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	resp, err := t.base().RoundTrip(req)
 
 	// TODO(jbd): Add status and attributes.
-	trace.EndSpan(ctx)
+	if err != nil {
+		trace.EndSpan(ctx)
+		return resp, err
+	}
+
+	// trace.EndSpan(ctx) will be invoked after
+	// resp.Body.Close() has been invoked.
+	resp.Body = &spanEndBody{rc: resp.Body, spanCtx: ctx}
 	return resp, err
 }
 
+// spanEndBody wraps a response.Body and invokes
+// trace.EndSpan on encountering io.EOF on reading
+// the body of the original response.
+type spanEndBody struct {
+	rc      io.ReadCloser
+	spanCtx context.Context
+
+	endSpanOnce sync.Once
+}
+
+var _ io.ReadCloser = (*spanEndBody)(nil)
+
+func (bpr *spanEndBody) Read(b []byte) (int, error) {
+	n, err := bpr.rc.Read(b)
+	if err == nil {
+		return n, nil
+	}
+
+	// Otherwise, time to end the span or set the status
+	switch err {
+	case io.EOF:
+		bpr.endSpan()
+	default:
+		// For all other errors, set the span status
+		// If the error has a status and code
+		// let's propagate those in the span
+		if ge, ok := err.(*googleapi.Error); ok {
```

Comment:
Don't import any Google API specific symbols into this package.

Reply:
Problem is that that is contrary to the hint in https://github.com/census-instrumentation/opencensus-go/blob/38381063fc2e146b7b20f389e35f505153a89a5b/trace/basetypes.go#L114-L121

otherwise we don't have another way of finding out the Status and Message unfortunately :(

Reply:
I think there is something I cannot follow. How can a raw HTTP request can return a googleapi.Error? There should be a layer in between that takes the raw http.Request body and converts it to a googleapi.Error object.

Reply:
To be clear, we won't do it in this package.

Reply:
Thank you @rakyll for breaking this down for me offline; I had a contrived assumption, I'll update this in a second.